core/src/main/java/io/questdb/cutlass/parquet/SerialParquetExporter.java

```diff
@@ -0,0 +1,409 @@
+                                if (version > maxVersion) {
+                                    maxVersion = version;
+                                    bestMatch = dirName.toString();
+                                }
+                            } catch (NumberFormatException ignored) {
+                            }
+                        }
+                        nameSink.clear();
+                    }
+                } while (ff.findNext(p) > 0);
+            } finally {
+                basePath.trimTo(plimit).$();
+                ff.findClose(p);
+            }
+        }
+
+        if (bestMatch != null) {
+            return bestMatch;
+        }
+        throw CairoException.nonCritical().put("no versioned parquet partition directory found for [basePartitionName=")
+                .put(basePartitionName).put(']');
+    }
+
+    private void moveAndOverwriteFiles() {
+        int srcPlen = tempPath.size();
+        int dstPlen = toParquet.size();
+        ff.iterateDir(tempPath.$(), (long name, int type) -> {
+            if (type == Files.DT_FILE) {
+                tempPath.trimTo(srcPlen).concat(name);
+                toParquet.trimTo(dstPlen).concat(name);
+                if (ff.exists(toParquet.$())) {
+                    LOG.info().$("overwriting existing file [path=").$(toParquet).$(']').$();
+                    ff.remove(toParquet.$());
+                }
+
+                if (ff.rename(tempPath.$(), toParquet.$()) != Files.FILES_RENAME_OK) {
+                    throw CopyExportException.instance(CopyExportRequestTask.Phase.CONVERTING_PARTITIONS, ff.errno())
+                            .put("could not copy export file [from=")
+                            .put(tempPath)
+                            .put(", to=")
+                            .put(toParquet)
+                            .put(", errno=").put(ff.errno()).put(']');
+                }
+            }
+        });
+    }
+
+    private void moveExportFiles(int tempBaseDirLen, String fileName) {
+        tempPath.trimTo(tempBaseDirLen);
+        if (!ff.exists(tempPath.$())) {
+            return;
+        }
+        toParquet.trimTo(0).concat(copyExportRoot).concat(fileName);
+        createDirsOrFail(ff, toParquet.slash(), configuration.getMkDirMode());
+        boolean destExists = ff.exists(toParquet.$());
+        if (!destExists) {
+            int moveResult = ff.rename(tempPath.$(), toParquet.$());
+            if (moveResult != Files.FILES_RENAME_OK) {
+                throw CopyExportException.instance(CopyExportRequestTask.Phase.CONVERTING_PARTITIONS, ff.errno())
+                        .put("could not rename export directory [from=")
+                        .put(tempPath)
+                        .put(", to=")
+                        .put(toParquet)
+                        .put(", errno=").put(ff.errno()).put(']');
+            }
+        } else {
+            moveAndOverwriteFiles();
+        }
+    }
+
+    StringSink getExportPath() {
```

Comment:
bad API, the return value is not meant to be used as a sink, it should be CharSequence